src/threads/blocks.rs

```diff
@@ -79,9 +96,10 @@ pub async fn index(state: Arc<State>, client: Client) -> Result<(), Error> {
         }
 
         let meta = BlockMeta::new(block_height, block.block_hash(), block.header.time);
-        state.set_hash_ts(&meta).await;
         db.update(&meta, utxo_spent, history_map, utxo_created)
+            .inspect_err(|e| log::error!("Error updating store: {e:?}"))
             .map_err(|_| Error::Other)?; // TODO
+        state.set_hash_ts(&meta).await;
```

Comment:
it seems it's called internally in `update()` also, should we remove it from here?